client/allochealth/tracker.go

```diff
@@ -278,12 +278,15 @@ func (t *Tracker) watchTaskEvents() {
 		latestStartTime := time.Time{}
 		for taskName, state := range alloc.TaskStates {
 			// One of the tasks has failed so we can exit watching
-			if state.Failed || (!state.FinishedAt.IsZero() && !t.lifecycleTasks[taskName]) {
+			if state.Failed || (!state.FinishedAt.IsZero() && (t.lifecycleTasks[taskName]) == "") {
 				t.setTaskHealth(false, true)
 				return
 			}
 
-			if state.State == structs.TaskStatePending {
+			// Ignore poststop since it will be pending until the main job exists
+			if (state.State == structs.TaskStatePending) &&
+				t.lifecycleTasks[taskName] != structs.TaskLifecycleHookPoststop {
```

Comment:
Do we ever want to consider poststop hooks as part of a deployment?  I don't think so in which case we should probably `if poststop { continue }` at the top of this loop? Otherwise I think if this code gets hit after poststop has started its started time will get taken into consideration?

Reply:
good question! @jazzyfresh @notnoop, I'm very unfamiliar with lifecycle hooks :D

Reply:
Agreed @schmichael. In both service & batch cases, deployments should watch for everything but the poststop tasks